esmvalcore/_citation.py

```diff
@@ -0,0 +1,229 @@
+
+    _save_citation_info(product_name, product_tags, json_urls, info_urls)
+
+
+def _save_citation_info(product_name, product_tags, json_urls, info_urls):
+    citation_entries = [ESMVALTOOL_PAPER]
+
+    # save CMIP6 url_info, if any
+    # save any refrences info that is not related to recipe or diagnostics
+    if info_urls:
+        with open(f'{product_name}_data_citation_info.txt', 'w') as file:
+            file.write('\n'.join(list(set(info_urls))))
+
+    # convert json_urls to bibtex entries
+    for json_url in json_urls:
+        cmip_citation = _collect_cmip_citation(json_url)
+        if cmip_citation:
+            citation_entries.append(cmip_citation)
+
+    # convert tags to bibtex entries
+    if REFERENCES_PATH and product_tags:
+        # make tags clean and unique
+        tags = list(set(_clean_tags(product_tags)))
+        for tag in tags:
+            citation_entries.append(_collect_bibtex_citation(tag))
+
+    with open(f'{product_name}_citation.bibtex', 'w') as file:
+        file.write('\n'.join(citation_entries))
+
+
+def _clean_tags(tags):
+    """Clean the tags that are recorded as str by provenance."""
+    pattern = re.compile(r'\w+')
+    return pattern.findall(str(tags))
+
+
+def _get_response(url):
+    """Return information from CMIP6 Data Citation service in json format."""
+    json_data = False
+    if url.lower().startswith('https'):
+        try:
+            response = requests.get(url)
+            if response.status_code == 200:
+                json_data = response.json()
+            else:
+                logger.info('Error in the CMIP json link: %s', url)
+        except IOError:
+            logger.info('Error in receiving the CMIP json file')
+    return json_data
+
+
+def _json_to_bibtex(data):
+    """Make a bibtex entry from CMIP6 Data Citation json data."""
+    url = 'url not found'
+    title = data.get('titles', ['title not found'])[0]
+    publisher = data.get('publisher', 'publisher not found')
+    year = data.get('publicationYear', 'publicationYear not found')
+    authors = 'creators not found'
+    doi = 'doi not found'
+
+    author_list = []
+    if data.get('creators', False):
+        author_list = [
+            item.get('creatorName', '') for item in data['creators']
+        ]
+    if author_list:
+        if author_list[0] == author_list[-1]:
+            authors = author_list[0]
+            if not authors:
+                authors = 'creatorName not found'
+        else:
+            authors = ' and '.join(author_list)
+
+    if data.get('identifier', False):
+        doi = data.get('identifier').get('id', 'doi not found')
+        url = f'https://doi.org/{doi}'
+
+    bibtex_entry = (
+        f'{"@misc{"}{url},\n\t'
+        f'url = {{{url}}},\n\t'
+        f'title = {{{title}}},\n\t'
+        f'publisher = {{{publisher}}},\n\t'
+        f'year = {year},\n\t'
+        f'author = {{{authors}}},\n\t'
+        f'doi = {{{doi}}},\n'
+        f'{"}"}\n'
+    )
```

Comment:
```suggestion
    bibtex_entry = textwrap.dedent(
        f"""
        @misc{{{url}
        \turl = {{{url}}},
        \ttitle = {{{title}}},
        \tpublisher = {{{publisher}}},
        \tyear = {year},
        \tauthor = {{{authors}}},
        \tdoi = {{{doi}}},
        }}
        """".lstrip()
    )
```
would be easier to read

Reply:
To stick to bibtex format, I removed the .lstrip() .

Reply:
Then the bibtex file will start with a newline, which is probably not what you want?